src/sxapi/base.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python
+# coding: utf8
+
+import requests
+
+PUBLIC_API_V2 = "https://api.smaxtec.com/api/v2"
+INTEGRATION_API_V2 = "https://api.smaxtec.com/integration/v2"
+
+
+class BaseAPI(object):
+    def __init__(self, base_url, email=None, password=None, api_key=None):
+        """Initialize a new base low level API client instance.
+        """
+        self.api_base_url = base_url.rstrip("/")
+        self.email = email
+        self.password = password
+        self.api_key = api_key
+        self._session_key = None
+        self._session = None
+
+    @property
+    def session(self):
+        """
+            Geneate a new HTTP session on the fly and login.
+        """
+        if not self._session:
+            self._session = requests.Session()
+        # check login
+        if not self._login():
+            raise ValueError("invalid login information")
+        return self._session
+
+    def to_url(self, path):
+        return f"{self.api_base_url}{path}"
+
+    def _login(self):
+        """
+            Login to the api with api key or the given credentials.
+        """
+        if self.api_key:
+            self._session_key = self.api_key
+            self._session.headers.update(
+                {"Authorization": "Bearer {}".format(self._session_key)})
```

Comment:
can we use fStrings for formatting? they are the standard since Python 3.6.

eg: `f"Authorization": "Bearer {self._session_key}"`

Reply:
nice find, thanks!